packages/firestore/src/api/timestamp.ts

```diff
@@ -92,4 +94,21 @@ export class Timestamp {
       ')'
     );
   }
+
+  // Overriding valueOf() allows Timestamp objects to be compared in JavaScript using the
+  // arithmetic comparison operators, such as < and >.
+  // https://github.com/firebase/firebase-js-sdk/issues/2632
+  //
+  // This method returns a string of the form <seconds>.<nanoseconds> where <seconds> is translated
+  // to have a non-negative value and both <seconds> and <nanoseconds> are left-padded with zeroes
+  // to be a consistent length. Strings with this format then have a lexiographical ordering that
+  // matches the relative ordering of the Timestamp objects whose valueOf() method returned them.
+  // The <seconds> translation is done to avoid having a leading negative sign (i.e. a leading '-'
```

Comment:
s/relative ordering of the Timestamp objects whose valueOf() method returned them/expected ordering.

Reply:
Done